vehicles/src/modules/common/dops.service.ts

```diff
@@ -308,4 +311,53 @@ export class DopsService {
       }
     }
   }
+
+  @LogAsyncMethodExecution()
+  async upload(
+    currentUser: IUserJWT,
+    companyId: number,
+    file: Express.Multer.File,
+    documentId?: Nullable<string>,
+  ): Promise<ReadFileDto> {
+    // Construct the URL for the request
+    let url = null;
+    if (!documentId) url = `${process.env.DOPS_URL}/dms/upload?`;
+    else url = `${process.env.DOPS_URL}/dms/upload/${documentId}?`;
```

Comment:
Can you braces to the if else?

Reply:
Also use `new URL()` to construct the url. Refer Line 401 in `frontend/permitsApi.ts`